lightwood/encoder/numeric/numeric.py

```diff
@@ -1,46 +1,63 @@
 import math
+from typing import Iterable, List, Union
 import torch
 import numpy as np
+from torch.types import Number
 from lightwood.encoder.base import BaseEncoder
 from lightwood.helpers.log import log
 from lightwood.helpers.general import is_none
+from lightwood.api.dtype import dtype
 
 
 class NumericEncoder(BaseEncoder):
+    """
+    The numeric encoder takes numbers (float or integer) and converts it into tensors of the form:
+    ``[is_none(number), is_positive(number), natural_log(abs(number)), number/absolute_mean]``
 
-    def __init__(self, data_type=None, is_target: bool = False, positive_domain: bool = False):
+    This representation is: ``[is_positive(number), natural_log(abs(number)), number/absolute_mean]]`` if encoding target values, since target values can't be none.
+
+    The ``absolute_mean`` is computed in the ``prepare`` method and is just the mean of the absolute values of all numbers feed to prepare (which are not none)
+
+    ``none`` stands for any number that is an actual python ``None`` value or any sort of non-numeric value (a string, nan, inf)
+    """ # noqa
+
+    def __init__(self, data_type: dtype = None, is_target: bool = False, positive_domain: bool = False):
+        """
+        :param data_type: The data type of the number (integer, float, quantity)
+        :param is_target: If we are encoding a target value or an input value (feature)
+        :param positive_domain: Forces the encoder to always output positive values
+        """
         super().__init__(is_target)
         self._type = data_type
         self._abs_mean = None
```

Comment:
what happens if _abs_mean is 0? This should in theory NEVER happen unless the input column are all 0s.

I can't recall in our stats-analysis if we throw away columns with extremely low entropy (i.e. columns where there is no difference at all in the values of each row. I recall some entropy calculation there for identifiers which are the opposite side). @George3d6 we should add this if it doesn't exist.

Reply:
The column will be removed before this happens because of the information content check.

Reply:
Also, if it's an input, it's a meaningless input, and even if we alter it such that it doesn't get removed, it will always be [0,0,0,0] (see try except in encoding inputs)... which is as well as having any other constant value to represent `0`